### DISCUSSION TITLE
Table-Export Fail

### DISCUSSION THREAD

**Discussion by RookieBrian** [dotnetcore/BootstrapBlazor#3938] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

BootstrapBlazor-8.7.3\src\BootstrapBlazor.Server中TablesExport导出PDF失败，EXCEL/CSV格式是没有问题的
![Uploading 1.png…]()

### Expected Behavior

可以像https://www.blazor.zone/table/export中一样正常导出PDF文件

### Steps To Reproduce

GitHUB原方案，无任何修改BootstrapBlazor-8.7.3\src\BootstrapBlazor.Server
点击页面中导出PDF按钮就无法导出，提示失败
![2](https://github.com/user-attachments/assets/9a014853-8a31-4390-a320-4c343c4b4db3)

### .NET Version

NET8.0

### Anything else?

[Log-20240725.log](https://github.com/user-attachments/files/16371351/Log-20240725.log)

**Comment by bb-auto** [comment]
@RookieBrian Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@RookieBrian 请帮忙右上角点一下 star 谢谢啦。然后是提供你不行的复现工程，我们没有办法通过你的描述进行任何修复工作，官网是可以下载的

**Comment by RookieBrian** [comment]
[https://github.com/RookieBrian/BootstrapBlazor.OnlyServerTable2PDF.git](https://github.com/RookieBrian/BootstrapBlazor.OnlyServerTable2PDF.git)
这个测试工程导出PDF就会报错

**Comment by khyjb1995** [comment]
Dear All

我遇到這個情景是在【**全新下載，第一次匯出PDF**】才會碰到

主因是缺少這兩個檔案【**chrome-win64.zip**】【**chrome-headless-shell-win64.zip**】
待下載完成後，即可正常匯出PDF了

情景説明：
第一次匯出PDF時，程式需要下載兩個檔案【**chrome-win64.zip**】【**chrome-headless-shell-win64.zip**】
在還未完成下載，又再次點擊匯出PDF系統會報錯，此時下載還在進行中
當下載完成後，會彈出儲存PDF的視窗
後續即可正常匯出PDF了

以下復現影片約2分鐘
![Table_Export_PDF_Failed](https://github.com/user-attachments/assets/b484fc01-ae9c-4d97-9a7f-19e1c5ca40f3)

**Reply by ArgoZhang** [reply, marked as answer]
@RookieBrian @khyjb1995 

### 解决方案

1. 增加 `BootstrapBlazor.Html2Pdf` 包
2. 注入服务

```
// 增加 Pdf 导出服务
builder.Services.AddBootstrapBlazorHtml2PdfService();
```


@khyjb1995 帮忙给项目一个星星可以吗？谢谢啦
